### ISSUE TITLE
`sigma_coordinates` on GPU but needed by data workers on CPU for `get_atmospheric_batch_data` 

### ISSUE THREAD

**Issue by jpdunc23** [ai2cm/ace#482]
This leads to the not super helpful error (tested with 109f6ed):

```
2024-11-01 12:51:10,104 - root - INFO - Opening data at /pscratch/sd/j/jpduncan/data/2024-09-20-cm4-1deg-8layer-trial-run/*.nc
Traceback (most recent call last):
  File "/global/cfs/cdirs/m4492/fme-env/109f6ed/lib/python3.10/multiprocessing/queues.py", line 244, in _feed
    obj = _ForkingPickler.dumps(obj)
  File "/global/cfs/cdirs/m4492/fme-env/109f6ed/lib/python3.10/multiprocessing/reduction.py", line 51, in dumps
    cls(buf, protocol).dump(obj)
  File "/global/cfs/cdirs/m4492/fme-env/109f6ed/lib/python3.10/site-packages/torch/multiprocessing/reductions.py", line 303, in reduce_tensor
    ) = storage._share_cuda_()
  File "/global/cfs/cdirs/m4492/fme-env/109f6ed/lib/python3.10/site-packages/torch/storage.py", line 1073, in _share_cuda_
    return self._untyped_storage._share_cuda_(*args, **kwargs)
RuntimeError: CUDA error: initialization error
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
```

@elynnwu has a working fix on https://github.com/ai2cm/full-model/pull/1305: https://github.com/ai2cm/full-model/pull/1305/commits/03f09c0cb53c62c8c119fff67b088b6a9a79b678

**Comment by jpdunc23** [comment]
@mcgibbon given this is related to ai2cm/ace#478 and your in-progress fix in https://github.com/ai2cm/full-model/pull/1310 it probably makes sense to address there. Elynn is going to push her fix that got https://github.com/ai2cm/full-model/pull/1305 working.

**Comment by jpdunc23** [comment]
https://github.com/ai2cm/full-model/commit/03f09c0cb53c62c8c119fff67b088b6a9a79b678 fixes the issue, but should be incorporated into ai2cm/full-model#1310.

**Comment by mcgibbon** [comment]
It sounds like an error is happening where the pickler can't send the sigma coordinates to other workers because it's on GPU. But won't putting the sigma coordinates on CPU cause an exception when the derive function is called, since the other data will be on GPU?

**Comment by elynnwu** [comment]
Good question. Are we just not hitting the derived func that uses sigma coordinates?

**Comment by brianhenn** [comment]
I think I'm hitting this on current main. See [experiment and logs](https://beaker-web.allen.ai/orgs/ai2/workspaces/ace/work/01JCERFED9VGY1Q4EXCPYX5DY1?taskId=01JCERFEDF0W8A402N5C4HFFE8&jobId=01JCERFEK9CQ2VV1WGA3ZBFW0B).

**Comment by oliverwm1** [comment]
Resolved by ai2cm/full-model#1322
